Place autoscroll marker on new line after previous empty line that is…

philip4096 opened the pull request:
… followed by a non-empty line to prevent marker from interfering with multiline latex equations. Issue #203

ashwinvis commented on ftplugin/markdown/instant-markdown.vim:
```diff
@@ -158,7 +158,17 @@ function! s:bufGetLines(bufnr)
+      return lines
+    endif
+
+    "From lines[row_num] find the previous empty line that is followed by a non-empty line, or the first line.
+    while ( row_num > 0 ) && !(lines[row_num] =~ '^\s*$' && !(lines[row_num + 1] =~ '^\s*$'))
+      let row_num += -1
+    endwhile
```
TBH, I think this while loop might be overkill. If this is the solution I wonder if we can replace this with a Vim search.

ashwinvis commented on ftplugin/markdown/instant-markdown.vim:
```diff
@@ -158,7 +158,17 @@ function! s:bufGetLines(bufnr)
+    endwhile
+    
+    "Insert new line with marker otherwise two paragraphs might be fused.
+    call insert(lines, '<a name="#marker" id="marker"></a>', row_num)
```
I wonder if we can solve this by adding `class="mathjax_ignore"` as documented here:

https://docs.mathjax.org/en/latest/options/document.html

ashwinvis replied:
@philip4096 you are right. It won't work.

philip4096 commented:
Thank you for the feedback. Using search seems like a good idea.

But I don’t see how the other option could be used to ignore a tag that is inside a latex equation, but perhaps I have misunderstood something. 

> 27 maj 2022 kl. 15:23 skrev Ashwin V. Mohanan ***@***.***>:
> 
> ﻿
> @ashwinvis commented on this pull request.
> 
> In ftplugin/markdown/instant-markdown.vim:
> 
> > +    "From lines[row_num] find the previous empty line that is followed by a non-empty line, or the first line.
> +    while ( row_num > 0 ) && !(lines[row_num] =~ '^\s*$' && !(lines[row_num + 1] =~ '^\s*$'))
> +      let row_num += -1
> +    endwhile
> TBH, I think this while loop might be overkill. If this is the solution I wonder if we can replace this with a Vim search.
> 
> In ftplugin/markdown/instant-markdown.vim:
> 
> > @@ -158,7 +158,17 @@ function! s:bufGetLines(bufnr)
>    if g:instant_markdown_autoscroll
>      " inject row marker
>      let row_num = max([0, line(".") - 5])
> -    let lines[row_num] = join([lines[row_num], '<a name="#marker" id="marker"></a>'], ' ')
> +    if len(lines) < 3
> +      return lines
> +    endif
> +
> +    "From lines[row_num] find the previous empty line that is followed by a non-empty line, or the first line.
> +    while ( row_num > 0 ) && !(lines[row_num] =~ '^\s*$' && !(lines[row_num + 1] =~ '^\s*$'))
> +      let row_num += -1
> +    endwhile
> +    
> +    "Insert new line with marker otherwise two paragraphs might be fused.
> +    call insert(lines, '<a name="#marker" id="marker"></a>', row_num)
> I wonder if we can solve this by adding class="mathjax_ignore" as documented here:
> 
> https://docs.mathjax.org/en/latest/options/document.html
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.

philip4096 commented:
I added a commit with while-loop replaced by search. Perhaps the regex pattern can be improved but it seems to work well. I also removed code to subtract 5 from current line number before starting search because I could not think of an easy way to use search together with that requirement. Also it seems like it isn't needed. What do you think?

ashwinvis commented:
It looks good. I will test it out.

ashwinvis commented:
Hey! Apologies for the delay. Tried it today and it looks good!
